### PR TITLE
[FLINK-29857][hive] Fix failure to connect to 'HiveServer2Endpoint' when using hive3 beeline

### PR DESCRIPTION
**Pull request by yuzelin**
## What is the purpose of the change
Using hive3's beeline to connect to SQL gateway with `HiveServer2Endpoint` will fail. This PR fix this bug.

## Brief change log
  - Handle unrecognized `TGetInfoType` value appropriately.
  - Add test.

## Verifying this change

This change added tests and can be verified as follows:

  - `HiveServer2EndpointITCase#testUnrecognizedGetInfoTypeWontCauseFailure`

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): **no**
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: **no**
  - The serializers: **no**
  - The runtime per-record code paths (performance sensitive): **no**
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: **no**
  - The S3 file system connector: **no**

## Documentation

  - Does this pull request introduce a new feature? **no**
  - If yes, how is the feature documented?

### CODE REVIEW

**Comment by yuzelin** [comment]
@flinkbot run azure

**Comment by yuzelin** [comment]
cc @fsk119

**Comment by yuzelin** [comment]
@flinkbot run azure

File: flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/endpoint/hive/HiveServer2EndpointITCase.java [hunk 1]
```diff
@@ -595,6 +598,27 @@ public void testGetInfo() throws Exception {
         }
     }
 
+    @Test
+    public void testUnrecognizedGetInfoTypeWontCauseFailure() throws Exception {
```
Comment (fsk119) [inline flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/endpoint/hive/HiveServer2EndpointITCase.java:602]: `testUnrecognizedGetInfoTypeWontCauseFailure` -> `testUnknownGetInfoType`

File: flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/endpoint/hive/HiveServer2EndpointITCase.java [hunk 2]
```diff
@@ -595,6 +598,27 @@ public void testGetInfo() throws Exception {
...
+        // When 'TGetInfoResp' is used and 'TGetInfoResp.infoValue' isn't set, hive service will
+        // close connection. Here using new 'TGetInfoType' CLI_ODBC_KEYWORDS added in hive3 sent by
+        // 'getSQLKeywords' to test whether it is handled appropriately.
+        Assumptions.assumeTrue(HiveVersionTestUtil.HIVE_310_OR_LATER);
+        try (Connection connection = ENDPOINT_EXTENSION.getConnection()) {
```
Comment (fsk119) [inline flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/endpoint/hive/HiveServer2EndpointITCase.java:607]: I think it's better to use Thrift Client to send GetInfoRequest directly and we can also test in Hive2 environment.

**Review by fsk119 — COMMENTED** [review]
Thanks for your contribution. I left some comments. 

Please also revert the last commit.

File: flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/endpoint/hive/HiveServer2EndpointITCase.java (hunk 1 above)
Comment (yuzelin) [reply flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/endpoint/hive/HiveServer2EndpointITCase.java:602]: Applied.

File: flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/endpoint/hive/HiveServer2EndpointITCase.java (hunk 2 above)
Comment (yuzelin) [reply flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/endpoint/hive/HiveServer2EndpointITCase.java:607]: Applied.

**Review by fsk119 — APPROVED** [review]
LGTM

**Pull request closed without merging**